phoenix-core/src/main/java/org/apache/phoenix/execute/BaseQueryPlan.java

```diff
@@ -204,8 +204,18 @@ public final ResultIterator iterator(final List<? extends SQLCloseable> dependen
         } else {
             ScanUtil.setTimeRange(scan, context.getScanTimeRange());
         }
-        
-        ScanUtil.setTenantId(scan, connection.getTenantId() == null ? null : connection.getTenantId().getBytes());
+        byte[] tenantIdBytes;
```

Comment:
I guess the question is why do we need to grab tenantIdBytes from the table (in the if {} block) now instead of just getting them from Connection.getTenantId().getBytes() like we did before?